broker/data-access-layer/iaas/test/iaas.AwsClient.spec.js

```diff
@@ -0,0 +1,293 @@
+              customkey: 'customvalue'
+            }
+          })
+          .catch(err => {
+            expect(err.message).to.eql('diskwaiterror');
+          });
+      });
+    });
+
+    describe('#getDiskMetadata', function () {
+      const diskId = 'sample-disk';
+      const response = {
+        Volumes: [{
+          VolumeId: diskId,
+          Size: '4',
+          AvailabilityZone: 'zone',
+          VolumeType: 'type',
+          Tags: [{
+            Key: 'k1',
+            Value: 'v1'
+          }, {
+            Key: 'k2',
+            Value: 'v2'
+          }]
+        }]
+      };
+      it('should fetch disk metadata for aws', function () {
+        const client = new AwsClient({
+          name: 'aws',
+          key: 'key',
+          keyId: 'keyId',
+          region: 'region'
+        });
+        const describeVolumesStub = sinon.stub(client.blockstorage, 'describeVolumes');
+        describeVolumesStub.withArgs({
+          VolumeIds: [diskId]
+        }).returns({
+          promise: () => Promise.resolve(response)
+        });
+        return client
+          .getDiskMetadata(diskId)
+          .then(disk => {
+            expect(disk.volumeId).to.eql(diskId);
+            expect(disk.size).to.eql('4');
+            expect(disk.zone).to.eql('zone');
+            expect(disk.type).to.eql('type');
+            expect(disk.extra).to.deep.equal({
+              type: 'type',
+              tags: {
+                k1: 'v1',
+                k2: 'v2'
+              }
+            });
+          });
+      });
+
+      it('should throw error while fetching disk metadata for aws', function () {
+        const client = new AwsClient({
+          name: 'aws',
+          key: 'key',
+          keyId: 'keyId',
+          region: 'region'
+        });
+        const describeVolumesStub = sinon.stub(client.blockstorage, 'describeVolumes');
+        describeVolumesStub.withArgs({
+          VolumeIds: [diskId]
+        }).returns({
+          promise: () => Promise.reject(new Error('diskerror'))
+        });
+        return client
+          .getDiskMetadata(diskId)
+          .catch(err => {
+            expect(err.message).to.eql('diskerror');
+          });
+      });
+    });
+
+    describe('#deleteSnapshot', function () {
+      it('bubbles up the error if aws throws error', function () {
+        const client = new AwsClient({
+          name: 'aws',
+          key: 'key',
+          keyId: 'keyId',
+          region: 'region'
+        });
+        const errorMessageExpected = 'fake-snap not found';
+        const deleteSnapshotStub = sinon.stub(client.blockstorage, 'deleteSnapshot');
+        deleteSnapshotStub.withArgs({
+          SnapshotId: 'fake-snap'
+        }).throws(Error(errorMessageExpected));
+        client
+          .deleteSnapshot('fake-snap')
+          .catch(err => expect(err.message).to.equal(errorMessageExpected));
+      });
+    });
+
+    describe('#ValidateParams', function () {
```

Comment:
Took inspiration from Unit Tests of `GcpClient` and tested the parameters passed to create the `AwsClient`